File: test_app/integration_tests/ios_mocks/VoxeetSDKMock/Classes/VideoPresentationServiceAsserts.swift
import Foundation

public class VideoPresentationServiceAsserts {
    
    private static var instance: VideoPresentationServiceAsserts?
    public static func create() {
        instance = .init()
    }
    
    var methodChannel: AssertsMethodChannel?
    
    public init() {
        assertsMethodChannelFactory.createAssertMethodChannel(
            forSdkAsserts: self,
            channelName: "IntegrationTesting.VideoPresentationServiceAsserts"
        )
    }
    
    private func assertStartArgs(args: [String: Any]) throws {
        let mockHasRun = VoxeetSDK.shared.videoPresentation.startHasRun
        try ifKeyExists(arg: args, key: "hasRun") { (hasRun: Bool) in
            try nativeAssertEquals(mockHasRun, hasRun)
        }
        
        let mockArgs = VoxeetSDK.shared.videoPresentation.startArgs
        try ifKeyExists(arg: args, key: "url") { (url: String) in
            try nativeAssertEquals(mockArgs?.absoluteString, url)
        }
    }
    
    private func assertStopArgs(args: [String: Any]) throws {
        let mockHasRun = VoxeetSDK.shared.videoPresentation.stopHasRun
        try ifKeyExists(arg: args, key: "hasRun") { (hasRun: Bool) in
            try nativeAssertEquals(mockHasRun, hasRun)
        }
    }
    
    private func assertPlayArgs(args: [String: Any]) throws {
        let mockHasRun = VoxeetSDK.shared.videoPresentation.playHasRun
        try ifKeyExists(arg: args, key: "hasRun") { (hasRun: Bool) in
            try nativeAssertEquals(mockHasRun, hasRun)
        }
    }
    
    private func assertPauseArgs(args: [String: Any]) throws {
        let mockHasRun = VoxeetSDK.shared.videoPresentation.pauseHasRun
        try ifKeyExists(arg: args, key: "hasRun") { (hasRun: Bool) in
            try nativeAssertEquals(mockHasRun, hasRun)
        }
        
        let mockArgs = VoxeetSDK.shared.videoPresentation.pauseArgs
        try ifKeyExists(arg: args, key: "timestamp") { (timestamp: Int) in
            try nativeAssertEquals(mockArgs, timestamp)
        }
    }
    
    private func assertSeekArgs(args: [String: Any]) throws {
        let mockHasRun = VoxeetSDK.shared.videoPresentation.seekHasRun
        try ifKeyExists(arg: args, key: "hasRun") { (hasRun: Bool) in
            try nativeAssertEquals(mockHasRun, hasRun)
        }
        
        let mockArgs = VoxeetSDK.shared.videoPresentation.seekArgs
        try ifKeyExists(arg: args, key: "timestamp") { (timestamp: Int) in
            try nativeAssertEquals(mockArgs, timestamp)
        }
    }
    
    private func assertStateArgs(args: [String: Any]) throws {
        let mockArgs = VoxeetSDK.shared.videoPresentation.state.rawValue
        try ifKeyExists(arg: args, key: "state") { (state: Int) in
            try nativeAssertEquals(mockArgs, state)
        }
    }
}

extension VideoPresentationServiceAsserts: SDKAsserts {
    
    public func runAssert(_ assert: String, args: [String: Any], result: (SDKAssertResult) -> Void) {
        do {
            switch assert {
                
            case "assertStartArgs":
                try assertStartArgs(args: args)
                
            case "assertStopArgs":
                try assertStopArgs(args: args)
                
            case "assertPlayArgs":
                try assertPlayArgs(args: args)
                
            case "assertPauseArgs":
                try assertPauseArgs(args: args)
                
            case "assertSeekArgs":
                try assertSeekArgs(args: args)
                
            case "assertStateArgs":
                try assertStateArgs(args: args)
                
            default:
                throw SDKAssertError.unknownAssert
            }
            result(.success(()))
        } catch {
            result(SDKAssertError.toResult(error: error))
        }
    }
}

File: test_app/integration_tests/android_mocks/src/main/java/com/voxeet/asserts/VideoPresentationServiceAsserts.java
package com.voxeet.asserts;

import com.voxeet.VoxeetSDK;
import com.voxeet.sdk.services.presentation.PresentationState;

import java.util.Map;

import io.dolby.asserts.AssertUtils;
import io.dolby.asserts.MethodDelegate;

public class VideoPresentationServiceAsserts implements MethodDelegate {
    @Override
    public void onAction(String methodName, Map<String, Object> args, Result result) {
        try {
            switch (methodName) {
                case "assertStartArgs":
                    assertStartArgs(args);
                    break;
                case "assertStopArgs":
                    assertStopArgs(args);
                    break;
                case "assertPlayArgs":
                    assertPlayArgs(args);
                    break;
                case "assertPauseArgs":
                    assertPauseArgs(args);
                    break;
                case "assertSeekArgs":
                    assertSeekArgs(args);
                    break;
                case "assertStateArgs":
                    assertStateArgs(args);
                    break;
                default:
                    result.error(new NoSuchMethodError());
                    return;
            }
        } catch (AssertionFailed exception) {
            result.failed(exception);
        } catch (Exception ex) {
            result.error( ex);
        }
        result.success();
    }

    @Override
    public String getName() {
        return "IntegrationTesting.VideoPresentationServiceAsserts";
    }

    public void assertStartArgs(Map<String, Object> args) throws AssertionFailed, KeyNotFoundException {
        boolean mockHasRun = VoxeetSDK.videoPresentation().startHasRun;
        if(args.containsKey("hasRun")) {
            AssertUtils.compareWithExpectedValue(mockHasRun, args.get("hasRun"), "startHasRun is incorrect");
        }

        String mockArgs = VoxeetSDK.videoPresentation().startArgs;
        if(args.containsKey("url")) {
            AssertUtils.compareWithExpectedValue(mockArgs, args.get("url"), "url is incorrect");
        }

    }

    public void assertStopArgs(Map<String, Object> args) throws AssertionFailed, KeyNotFoundException {
        boolean mockHasRun = VoxeetSDK.videoPresentation().stopHasRun;
        if(args.containsKey("hasRun")) {
            AssertUtils.compareWithExpectedValue(mockHasRun, args.get("hasRun"), "stopHasRun is incorrect");
        }
    }

    public void assertPlayArgs(Map<String, Object> args) throws AssertionFailed, KeyNotFoundException {
        boolean mockHasRun = VoxeetSDK.videoPresentation().playHasRun;
        if(args.containsKey("hasRun")) {
            AssertUtils.compareWithExpectedValue(mockHasRun, args.get("hasRun"), "playHasRun is incorrect");
        }
    }

    public void assertPauseArgs(Map<String, Object> args) throws AssertionFailed, KeyNotFoundException {
        boolean mockHasRun = VoxeetSDK.videoPresentation().pauseHasRun;
        if(args.containsKey("hasRun")) {
            AssertUtils.compareWithExpectedValue(mockHasRun, args.get("hasRun"), "pauseHasRun is incorrect");
        }

        int mockArgs = (int) VoxeetSDK.videoPresentation().pauseArgs;
        if(args.containsKey("timestamp")) {
            AssertUtils.compareWithExpectedValue(mockArgs, args.get("timestamp"), "timestamp is incorrect");
        }
    }

    public void assertSeekArgs(Map<String, Object> args) throws AssertionFailed, KeyNotFoundException {
        boolean mockHasRun = VoxeetSDK.videoPresentation().seekHasRun;
        if(args.containsKey("hasRun")) {
            AssertUtils.compareWithExpectedValue(mockHasRun, args.get("hasRun"), "seekHasRun is incorrect");
        }

        int mockArgs = (int) VoxeetSDK.videoPresentation().seekArgs;
        if(args.containsKey("timestamp")) {
            AssertUtils.compareWithExpectedValue(mockArgs, args.get("timestamp"), "timestamp is incorrect");
        }
    }

    public void assertStateArgs(Map<String, Object> args) throws AssertionFailed, KeyNotFoundException {
        int mockArgs = VoxeetSDK.videoPresentation().getCurrentPresentation().state.value();
        if(args.containsKey("state")) {
            AssertUtils.compareWithExpectedValue(mockArgs, args.get("state"), "state is incorrect");
        }
    }
}
